package.json

```diff
@@ -1,61 +1,80 @@
 {
   "name": "tinypool",
+  "type": "module",
   "version": "0.8.4",
+  "packageManager": "pnpm@9.0.6",
   "description": "A minimal and tiny Node.js Worker Thread Pool implementation, a fork of piscina, but with fewer features",
-  "type": "module",
-  "main": "./dist/esm/index.js",
-  "module": "./dist/esm/index.js",
+  "license": "MIT",
+  "homepage": "https://github.com/tinylibs/tinypool#readme",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/tinylibs/tinypool.git"
+  },
+  "bugs": {
+    "url": "https://github.com/tinylibs/tinypool/issues"
+  },
+  "keywords": [
+    "fast",
+    "worker threads",
+    "thread pool"
+  ],
+  "exports": {
+    ".": {
+      "types": "./dist/index.d.ts",
+      "default": "./dist/index.js"
+    },
+    "./entry/utils": {
+      "types": "./dist/entry/utils.d.ts",
+      "default": "./dist/entry/utils.js"
+    }
+  },
+  "main": "./dist/index.js",
+  "module": "./dist/index.js",
   "types": "./dist/index.d.ts",
+  "typesVersions": {
+    "*": {
+      "entry/utils": [
+        "./dist/entry/utils.d.ts"
+      ],
+      "*": [
+        "./dist/index.d.ts"
+      ]
+    }
+  },
   "files": [
-    "dist/**"
+    "dist"
   ],
-  "packageManager": "pnpm@8.4.0",
+  "engines": {
+    "node": "^18.0.0 || >=20.0.0"
+  },
   "scripts": {
     "test": "vitest",
     "dev": "tsup --watch",
     "build": "tsup",
     "publish": "clean-publish",
     "prepare": "husky install"
   },
-  "repository": {
-    "type": "git",
-    "url": "git+https://github.com/tinylibs/tinypool.git"
-  },
-  "keywords": [
-    "fast",
-    "worker threads",
-    "thread pool"
-  ],
-  "license": "MIT",
   "devDependencies": {
     "@types/concat-stream": "^1.6.1",
-    "@types/node": "^15.0.1",
-    "@typescript-eslint/eslint-plugin": "^4.22.0",
-    "@typescript-eslint/parser": "^4.22.0",
+    "@types/node": "^20.12.8",
+    "@typescript-eslint/eslint-plugin": "^7.8.0",
+    "@typescript-eslint/parser": "^7.8.0",
     "abort-controller": "^3.0.0",
     "clean-publish": "^3.4.4",
     "concat-stream": "^2.0.0",
     "esbuild": "^0.14.5",
     "esbuild-register": "^3.2.1",
-    "eventemitter-asyncresource": "^1.0.0",
     "gen-esm-wrapper": "^1.1.1",
     "husky": "^7.0.4",
     "nano-staged": "^0.5.0",
     "prettier": "^2.5.1",
     "regenerator-runtime": "^0.13.9",
     "snazzy": "^9.0.0",
-    "tsup": "^5.11.6",
-    "typescript": "4.3.x",
+    "tsup": "^8.0.2",
+    "typescript": "^5.4.5",
     "vite": "^5.2.10",
     "vitest": "^1.5.0"
   },
```

Comment:
Looks like there are plenty of unused dependencies. We should be able to remove all of these as well:

- `@types/concat-stream`
- `concat-stream`
- `@typescript-eslint/eslint-plugin` (ESLint is not set for this repo)
- `@typescript-eslint/eslint-parser`
- `abort-controller`
- `esbuild`
- `esbuild-register`
- `gen-esm-wrapper`
- `regenerator-runtime`
- `snazzy`